vm/src/builtins/type.rs

```diff
@@ -405,6 +405,35 @@ impl PyType {
             .unwrap_or_else(|| vm.ctx.new_str(self.name().deref()).into())
     }
 
+    #[pyproperty(magic, setter)]
+    fn set_qualname(&self, value: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
+        if value.is(&vm.ctx.none()) {
+            return Err(vm.new_type_error(format!(
+                "cannot delete '{}' attribute of immutable type '{}'",
+                "__qualname__",
+                self.name()
+            )));
+        };
```

Comment:
```suggestion
    fn set_qualname(&self, value: Option<PyObjectRef>, vm: &VirtualMachine) -> PyResult<()> {
        let value = value.ok_or_else(|| vm.new_type_error(format!(
            "cannot delete '{}' attribute of immutable type '{}'",
            "__qualname__",
            self.name()
        )))?;
```
We can also pass `Option<PyObjectRef>` for separating whether it is a setter call or a delete call.

Reply:
Oh, I think we don't have proper `PySetterFunc` to handle deleter case

Reply:
To distinguish setting None and delete, #4010 is required.

Reply:
Oh I missed that setting None also possible 😂

Reply:
Oh.... #4010 has not been merged yet, so I will correct it when it is merged. Thank you.

Reply:
#4010 is done

Reply:
I checked it. Thank you !